src/test/java/com/karankumar/bookproject/ui/book/BookFormTest.java

```diff
@@ -599,6 +599,26 @@ void shouldAddBookToDatabaseWhenSaveEventIsCalled() {
         correctBookAttributesPresent(TO_READ, bookInDatabase);
     }
 
+    @Test
+    void shouldAddBooksToDatabaseWhenSaveEventIsCalled_withoutReplacingExistingBook() {
+        // given
+        bookForm = createBookForm(TO_READ, false);
+        bookForm.addListener(BookForm.SaveEvent.class, event -> bookService.save(event.getBook()));
+        bookForm.saveButton.click();
+
+        // when
+        bookTitle = "someOtherBook";
```

Comment:
I wouldn't recommend changing the name of the global variable and not making it `final` without resetting it back (or some other way). 

We use the `bookTitle` in assertions, so it's best to keep it to a value that's known (in case the tests run in a different order).

Reply:
Done